src/services/CertificateGenerationService.ts

```diff
@@ -177,6 +174,60 @@ class CertificateGenerationService {
             });
     }
 
+    /**
+     * Method to check if Welsh translation is enabled.
+     * @param flags
+     * @returns boolean
+     */
+    public isGlobalWelshFlagEnabled(flags: any): boolean {
+        if (!flags.welshTranslation.enabled) {
+            console.warn(`Unable to translate any test results: global Welsh flag disabled.`);
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Method to check if Welsh translation is enabled for the given test type.
+     * @param flags
+     * @param testResult
+     * @returns boolean
+     */
+    public isTestResultFlagEnabled(flags: any, testResult: string): boolean {
+        let shouldTranslate: boolean = false;
+        switch (testResult) {
+            case TEST_RESULTS.PRS:
+                shouldTranslate = flags.welshTranslation.translatePrsTestResult ? true : false;
+                break;
+            case TEST_RESULTS.PASS:
+                shouldTranslate = flags.welshTranslation.translatePassTestResult ? true : false;
+                break;
+            case TEST_RESULTS.FAIL:
+                shouldTranslate = flags.welshTranslation.translateFailTestResult ? true : false;
+                break;
+            default:
+                console.warn("Translation not required for this test result type.");
+                return shouldTranslate;
+        }
+        if (!shouldTranslate) {
+            console.warn(`Unable to translate for test result: ${testResult} flag disabled`);
+        }
+        return shouldTranslate;
+    }
+
+    /**
+     * Method to check if the test result requires Welsh translation
+     * @param testResult
```

Comment:
Can you update this doc block please, to include the updated parameter. 

Also, to avoid confusion may be worth calling it testStationPNumber as that is what we are used to, rather than testResultPNumber.